src/lib/utils/math.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Get decimal places of float
+ * @param {number} a - float number
+ * @returns {number} The number of decimal places of the float number.
+ * @example
+ * floatPrecision(54.6545) // 4
+ */
+export function floatPrecision(a: number): number {
+  if (!isFinite(a)) return 0
+  var e = 1,
+    p = 0
+  while (Math.round(a * e) / e !== a) {
+    e *= 10
+    p++
+  }
+  return p
+}
+
+/**
+ * Round a number to the nearest step
+ * @param {number} value - The number to round.
+ * @param {number} step - The step to round to.
+ * @returns {number} The rounded number.
+ * @example
+ * roundToStep(54.6545, 0.1) // 54.7
+ */
+export function roundToStep(value: number, step: number): number {
```

Comment:
What if the step is 0.2? It will round to 0.1, right? Is that intended behaviour? If so, maybe this should be called `roundToStepPrecision` or something?

Reply:
Renamed to roundToStepPrecision